### PR TITLE
Show error in CI pipeline when deployment fails

### PR DESCRIPTION
**Pull request by jsugarman**
#### What
Fail deployment overtly if "pods" do not reach "readiness"

#### Ticket

[CBO-1334](https://dsdmoj.atlassian.net/browse/CBO-1334)

#### Why
Currently deployment is green as long as the Kubernetes apply commands are successful, but the pods may still fail to start the application up and various other scenarios could prevent the deployment actually succeeding.

### CODE REVIEW

File: .circleci/deploy.sh [hunk 1]
```diff
@@ -10,6 +10,12 @@ function _circleci_deploy() {
     deploy.sh dev
     "
 
+  # exit when any command fails, keep track of the last
+  # executed command and output error message before exiting
+  # https://intoli.com/blog/exit-on-errors-in-bash-scripts/
```
Comment (jsugarman) [inline .circleci/deploy.sh]: remove

File: .circleci/deploy.sh [hunk 2]
```diff
@@ -55,6 +61,8 @@ function _circleci_deploy() {
   echo "${GIT_CRYPT_KEY}" | base64 -d > git-crypt.key
   git-crypt unlock git-crypt.key
 
+  echo $(kubectl version)
```
Comment (jsugarman) [inline .circleci/deploy.sh]: remove

File: .circleci/deploy.sh [hunk 3]
```diff
@@ -85,6 +89,10 @@ function _circleci_deploy() {
 
   kubectl annotate deployments/claim-for-crown-court-defence kubernetes.io/change-cause="$(date +%Y-%m-%dT%H:%M:%S%z) - deploying: $docker_image_tag via CircleCI"
   kubectl annotate deployments/claim-for-crown-court-defence-worker kubernetes.io/change-cause="$(date +%Y-%m-%dT%H:%M:%S%z) - deploying: $docker_image_tag via CircleCI"
+
+  # wait for rollout to succeed or fail/timeout
+  kubectl rollout status deployments/claim-for-crown-court-defence
```
Comment (kmahern) [inline .circleci/deploy.sh:94]: Not sure when this will timeout?

Comment (jsugarman) [reply .circleci/deploy.sh:94]: default is 10 minutes. to have more control we need to upgrade kubectl 1.12+ client or set `spec.progressDeadlineSeconds`

Comment (jsugarman) [reply .circleci/deploy.sh:94]: these are other issues that i will need to address in another PR.

**Review by kmahern — APPROVED** [review]
lgtm 👍

**Pull request merged**